internal/apmhostutil/container_linux.go

```diff
@@ -46,12 +46,12 @@ var (
 	kubepodsRegexp = regexp.MustCompile(
 		"" +
 			`(?:^/kubepods[\S]*/pod([^/]+)/$)|` +
-			`(?:^/kubepods\.slice/kubepods-[^/]+\.slice/kubepods-[^/]+-pod([^/]+)\.slice/$)`,
+			`(?:kubepods[^/]*-pod([^/]+)\.slice/)`,
 	)
 
 	containerIDRegexp = regexp.MustCompile(
 		"^" +
-			"[[:xdigit:]]{64}|" +
+			"[[:xdigit:]]{64}$|" +
 			"[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4,}" +
 			"$",
```

Comment:
I think the original intention was that the "^" and "$" were to apply regardless of which of the other lines matched. Possibly there was originally a non-capturing group surrounding those options. Anyway, I think we should add "^" and "$" to each of the lines, so it's:

```go
	containerIDRegexp = regexp.MustCompile(
			"^[[:xdigit:]]{64}$|" +
			"^[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4,}$",
	)
```

WDYT?

Reply:
I agree, this should also be more in line with the spec 👍